extensions/jdbc/jdbc-mysql/jdbc-mysql-deployment/src/main/java/io/quarkus/jdbc/mysql/deployment/MySQLDBJDBCReflections.java

```diff
@@ -0,0 +1,99 @@
+import com.mysql.cj.exceptions.PropertyNotModifiableException;
+import com.mysql.cj.exceptions.RSAException;
+import com.mysql.cj.exceptions.SSLParamsException;
+import com.mysql.cj.exceptions.StatementIsClosedException;
+import com.mysql.cj.exceptions.StreamingNotifiable;
+import com.mysql.cj.exceptions.UnableToConnectException;
+import com.mysql.cj.exceptions.UnsupportedConnectionStringException;
+import com.mysql.cj.exceptions.WrongArgumentException;
+import com.mysql.cj.protocol.AsyncSocketFactory;
+import com.mysql.cj.protocol.NamedPipeSocketFactory;
+import com.mysql.cj.protocol.SocksProxySocketFactory;
+import com.mysql.cj.protocol.StandardSocketFactory;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+
+public final class MySQLDBJDBCReflections {
+
+    @BuildStep
+    void registerDriverForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, com.mysql.cj.jdbc.Driver.class.getName()));
+        reflectiveClass.produce(
+                new ReflectiveClassBuildItem(false, false, com.mysql.cj.conf.url.FailoverConnectionUrl.class.getName()));
+        reflectiveClass
+                .produce(new ReflectiveClassBuildItem(false, false, com.mysql.cj.conf.url.SingleConnectionUrl.class.getName()));
+        reflectiveClass.produce(
+                new ReflectiveClassBuildItem(false, false, com.mysql.cj.conf.url.LoadbalanceConnectionUrl.class.getName()));
+        reflectiveClass.produce(
+                new ReflectiveClassBuildItem(false, false, com.mysql.cj.conf.url.ReplicationConnectionUrl.class.getName()));
+        reflectiveClass.produce(
+                new ReflectiveClassBuildItem(false, false, com.mysql.cj.conf.url.XDevAPIConnectionUrl.class.getName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false,
+                com.mysql.cj.jdbc.ha.LoadBalancedAutoCommitInterceptor.class.getName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, com.mysql.cj.log.StandardLogger.class.getName()));
+    }
+
+    @BuildStep
+    void registerSocketFactoryForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, NamedPipeSocketFactory.class.getName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, StandardSocketFactory.class.getName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, SocksProxySocketFactory.class.getName()));
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, AsyncSocketFactory.class.getName()));
+    }
+
+    @BuildStep
+    void registerExceptionsForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
```

Comment:
I pushed an upstream PR https://github.com/mysql/mysql-connector-j/pull/43 to reduce some of these. The PR has been moved in the internal queue https://github.com/mysql/mysql-connector-j/pull/43#issuecomment-529560017, hope to see it merged.